src/test/java/RequestParserTest.java

```diff
@@ -0,0 +1,62 @@
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import static http.request.RequestParser.*;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RequestParserTest {
+
+
+    @Test
+    public void checkThatAStartLineDataIsSet() {
+        setRequestType("GET / HTTP/1.1");
+
+        assertEquals(getRequestType(), "GET / HTTP/1.1".trim());
+
+    }
+
+    @Test
+    public void checkThatHeadersAreSet() {
+        setHeaders("text/html");
+
+        assertEquals(getHeaders(), headers);
+
+    }
+
+    @Test
+    public void checkThatRequestBodyWasSet() {
+        setRequestBody("hello".getBytes());
+
+        assertEquals(getRequestBody(), "hello");
+
+    }
+
+    @Test
+    public void checkThatStatusIsUpdated() {
+        setStatus("200", "well done :)");
+
+        assertEquals(getStatusCode(), "200");
+        assertEquals(getStatusMessage(), "well done :)");
+    }
+
+    @Test
+    public void checkThatFormattedResponseIsReturned() {
+        setRequestType("GET / HTTP/1.1");
+        setHeaders("text/html");
+        System.out.println(requestBody + "*****************");
+        setRequestBody("hello".getBytes(StandardCharsets.UTF_8));
+        setStatus("200", "well done :)");
+        System.out.println(requestBody + "*****************");
```

Comment:
Are the lines being printed out here useful for the test, or were they put there as you were putting the test together to help you see what was happening?

If it's the latter I'd suggest removing them once the test works as you want it to.